plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -91,6 +97,151 @@ export class GitLabClient {
     return this.pagedRequest(`/groups`, options);
   }
 
+  async listSaasUsers(groupPath: string): Promise<PagedResponse<GitLabUser>> {
+    const items: GitLabUser[] = [];
+    let hasNextPage: boolean = false;
+    let endCursor: string | null = null;
+
+    do {
+      const response: GitLabSaasUsersResponse = await fetch(
+        `${this.config.baseUrl}/api/graphql`,
+        {
+          method: 'POST',
+          headers: {
+            ...getGitLabRequestOptions(this.config).headers,
+            ['Content-Type']: 'application/json',
+          },
+          body: JSON.stringify({
+            variables: { group: groupPath, endCursor },
+            query: `query listSaasUsers($group: ID!, $endCursor: String) {
+              group(fullPath: $group) {
+                groupMembers(first: 100, relations: [DESCENDANTS], after: $endCursor) {
```

Comment:
If a user is a member of the referenced group but not any of its descendants, I think they should still be included
```suggestion
                groupMembers(first: 100, relations: [DIRECT, DESCENDANTS], after: $endCursor) {
```

Reply:
Makes sense, done

Reply:
Looks like the GraphQL API expects of one [`GroupMemberRelation`](https://docs.gitlab.com/ee/api/graphql/reference/#groupmemberrelation), so switching to `DIRECT` instead